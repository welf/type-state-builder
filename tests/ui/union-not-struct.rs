use type_state_builder::TypeStateBuilder;

// This should be an error: unions are not supported, only structs
#[derive(TypeStateBuilder)]
union NotAStruct {
    field1: u32,
    field2: f32,
}
fn main() {}
