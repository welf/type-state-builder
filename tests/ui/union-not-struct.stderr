error: TypeStateBuilder only supports structs, not unions
 --> tests/ui/union-not-struct.rs:5:1
  |
5 | / union NotAStruct {
6 | |     field1: u32,
7 | |     field2: f32,
8 | | }
  | |_^
